src/smt/solver_engine.cpp

```diff
@@ -2131,13 +2139,30 @@ void SolverEngine::setOption(const std::string& key,
       {
         d_safeOptsSetRegularOption = true;
         d_safeOptsRegularOption = key;
+        d_safeOptsRegularOptionValue = value;
+        d_safeOptsSetRegularOptionToDefault = (getOption(key) == value);
       }
       else
       {
         // option exception
         std::stringstream ss;
-        ss << "cannot set two regular options (" << key << " and "
-           << d_safeOptsRegularOption << ") when safeOptions is true";
+        ss << "cannot set two regular options (" << d_safeOptsRegularOption
+           << " and " << key << ") when safeOptions is true.";
+        // similar to above, if setting to default value for either of the
+        // regular options.
+        for (size_t i = 0; i < 2; i++)
+        {
+          std::string rkey = i == 0 ? d_safeOptsRegularOption : key;
+          std::string rvalue = i == 0 ? d_safeOptsRegularOptionValue : value;
```

Comment:
```suggestion
          const std::string& rkey = i == 0 ? d_safeOptsRegularOption : key;
          const std::string& rvalue = i == 0 ? d_safeOptsRegularOptionValue : value;
```
A copy shouldn't be necessary here.